### ISSUE TITLE
txg_sync hung in D state, any process that tries to write on the pool also hangs, but reads seem OK

### ISSUE THREAD

**Issue by DurvalMenezes** [zfsonlinux/zfs#4936]
Hello,

As per the subject above. Issue started after machine resumed from suspend, but this seems to be a coincidence as I do not see any signs of ACPI trouble.

System is a Lenovo W520 notebook running Springdale 6.8 (RHEL6 clone, very similar to CentOS) and ZFS version 0.6.5.6-1 installed via yum from the official ZFS-On-Linux repos. 

Pool is a mirror on two internal harddisks (sda + sdb), with log and cache on an internal SSD (sdc). All devices are working perfectly (nothing on the logs, and dd'ing from them to /dev/null completes normally, and "zfs pool" shows everything as OK). 

Will attach syslog (yes, system is logging normally as /var/log is on another, non-zfs filesystem) and also zpool status output.

Cheers, 
 -- 
   Durval Menezes.

**Comment by DurvalMenezes** [comment]
Hello,

Relevant parts of syslog and related output follows:

0) System came back from suspend normally, and crond even started zfs-auto-snapshot.sh with apparently no issues:
====================================================[
Aug  5 05:28:54 localhost kdump: kexec: loaded kdump kernel
Aug  5 05:28:54 localhost kdump: started up
Aug  5 05:30:01 localhost crond[29152]: (*system*) NULL security context for user, but SELinux in p
ermissive mode, continuing ()
Aug  5 05:30:01 localhost crond[29151]: (*system*) NULL security context for user, but SELinux in p
ermissive mode, continuing ()
Aug  5 05:30:01 localhost CROND[29155]: (root) CMD (/usr/lib64/sa/sa1 1 1)
Aug  5 05:30:01 localhost CROND[29156]: (root) CMD ((. /etc/sysconfig/zfs-auto-snapshot; /usr/local
/sbin/zfs-auto-snapshot ${ZAS_COMMON_OPTS} --label=frequent --keep=${ZAS_FREQUENT_KEEP}  ${ZAS_FREQ
UENT_FS}))
====================================================] 

On the other hand, the "zfs snapshot" subprocess, called by the above script, apparently hang; according to "ps auwx: 

root     29181  0.0  0.0 127616  1596 ?        D    05:30   0:00 zfs snapshot -o com.sun:auto-snapshot-desc - -r pool01/usr2 zfs-auto-snap_frequent-2016-08-05-0830

1) Here's the first sign of trouble on syslog, the infamous "task ... blocked for more than ... seconds":
====================================================[ 
Aug  5 05:30:44 localhost kernel: INFO: task txg_sync:3268 blocked for more than 600 seconds.
Aug  5 05:30:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:30:44 localhost kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Aug  5 05:30:44 localhost kernel: txg_sync      D 0000000000000000     0  3268      2 0x00000000
Aug  5 05:30:44 localhost kernel: ffff8807fb5dbb60 0000000000000046 0000000000000000 ffff88080822e338
Aug  5 05:30:44 localhost kernel: 0000000000000000 0000000000000000 000030527f201a2f ffffffff810672c2
Aug  5 05:30:44 localhost kernel: ffff8807fb5dbb20 0000000103263c30 ffff8807fc87c5f8 ffff8807fb5dbfd8
Aug  5 05:30:44 localhost kernel: Call Trace:
Aug  5 05:30:44 localhost kernel: [<ffffffff810672c2>] ? default_wake_function+0x12/0x20
Aug  5 05:30:44 localhost kernel: [<ffffffff81538d33>] io_schedule+0x73/0xc0
Aug  5 05:30:44 localhost kernel: [<ffffffffa06acdf1>] cv_wait_common+0xb1/0x130 [spl]
Aug  5 05:30:44 localhost kernel: [<ffffffff810a1460>] ? autoremove_wake_function+0x0/0x40
Aug  5 05:30:44 localhost kernel: [<ffffffffa06ace88>] __cv_wait_io+0x18/0x20 [spl]
Aug  5 05:30:44 localhost kernel: [<ffffffffa07b0c1b>] zio_wait+0x10b/0x1e0 [zfs]
Aug  5 05:30:44 localhost kernel: [<ffffffffa0737923>] dsl_pool_sync+0xb3/0x440 [zfs]
Aug  5 05:30:44 localhost kernel: [<ffffffffa074ec3e>] spa_sync+0x35e/0xb10 [zfs]
Aug  5 05:30:44 localhost kernel: [<ffffffff81059939>] ? __wake_up_common+0x59/0x90
Aug  5 05:30:44 localhost kernel: [<ffffffff8105e173>] ? __wake_up+0x53/0x70
Aug  5 05:30:44 localhost kernel: [<ffffffff81014a19>] ? read_tsc+0x9/0x10
Aug  5 05:30:44 localhost kernel: [<ffffffffa0765358>] txg_sync_thread+0x3d8/0x670 [zfs]
Aug  5 05:30:44 localhost kernel: [<ffffffffa0764f80>] ? txg_sync_thread+0x0/0x670 [zfs]
Aug  5 05:30:44 localhost kernel: [<ffffffffa0764f80>] ? txg_sync_thread+0x0/0x670 [zfs]
Aug  5 05:30:44 localhost kernel: [<ffffffffa06a8238>] thread_generic_wrapper+0x68/0x80 [spl]
Aug  5 05:30:44 localhost kernel: [<ffffffffa06a81d0>] ? thread_generic_wrapper+0x0/0x80 [spl]
Aug  5 05:30:44 localhost kernel: [<ffffffff810a0fce>] kthread+0x9e/0xc0
Aug  5 05:30:44 localhost kernel: [<ffffffff8100c28a>] child_rip+0xa/0x20
Aug  5 05:30:44 localhost kernel: [<ffffffff810a0f30>] ? kthread+0x0/0xc0
Aug  5 05:30:44 localhost kernel: [<ffffffff8100c280>] ? child_rip+0x0/0x20
====================================================]

2) The fun continues (the "vim" process below was trying to save a *LONG* file I was editing... I hope I can get it back with vim's recover function after I reboot the machine):
====================================================[
Aug  5 05:40:44 localhost kernel: INFO: task txg_sync:3268 blocked for more than 600 seconds.
Aug  5 05:40:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:40:44 localhost kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Aug  5 05:40:44 localhost kernel: txg_sync      D 0000000000000000     0  3268      2 0x00000000
Aug  5 05:40:44 localhost kernel: ffff8807fb5dbb60 0000000000000046 0000000000000000 ffff88080822e338
Aug  5 05:40:44 localhost kernel: 0000000000000000 0000000000000000 000030527f201a2f ffffffff810672c2
Aug  5 05:40:44 localhost kernel: ffff8807fb5dbb20 0000000103263c30 ffff8807fc87c5f8 ffff8807fb5dbfd8
Aug  5 05:40:44 localhost kernel: Call Trace:
Aug  5 05:40:44 localhost kernel: [<ffffffff810672c2>] ? default_wake_function+0x12/0x20
Aug  5 05:40:44 localhost kernel: [<ffffffff81538d33>] io_schedule+0x73/0xc0
Aug  5 05:40:44 localhost kernel: [<ffffffffa06acdf1>] cv_wait_common+0xb1/0x130 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffff810a1460>] ? autoremove_wake_function+0x0/0x40
Aug  5 05:40:44 localhost kernel: [<ffffffffa06ace88>] __cv_wait_io+0x18/0x20 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07b0c1b>] zio_wait+0x10b/0x1e0 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa0737923>] dsl_pool_sync+0xb3/0x440 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa074ec3e>] spa_sync+0x35e/0xb10 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffff81059939>] ? __wake_up_common+0x59/0x90
Aug  5 05:40:44 localhost kernel: [<ffffffff8105e173>] ? __wake_up+0x53/0x70
Aug  5 05:40:44 localhost kernel: [<ffffffff81014a19>] ? read_tsc+0x9/0x10
Aug  5 05:40:44 localhost kernel: [<ffffffffa0765358>] txg_sync_thread+0x3d8/0x670 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa0764f80>] ? txg_sync_thread+0x0/0x670 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa0764f80>] ? txg_sync_thread+0x0/0x670 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa06a8238>] thread_generic_wrapper+0x68/0x80 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffffa06a81d0>] ? thread_generic_wrapper+0x0/0x80 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffff810a0fce>] kthread+0x9e/0xc0
Aug  5 05:40:44 localhost kernel: [<ffffffff8100c28a>] child_rip+0xa/0x20
Aug  5 05:40:44 localhost kernel: [<ffffffff810a0f30>] ? kthread+0x0/0xc0
Aug  5 05:40:44 localhost kernel: [<ffffffff8100c280>] ? child_rip+0x0/0x20
Aug  5 05:40:44 localhost kernel: INFO: task AioMgr13-F:6323 blocked for more than 600 seconds.
Aug  5 05:40:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:40:44 localhost kernel: INFO: task AioMgr13-F:6323 blocked for more than 600 seconds.
Aug  5 05:40:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:40:44 localhost kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Aug  5 05:40:44 localhost kernel: AioMgr13-F    D 0000000000000003     0  6323   6106 0x00000000
Aug  5 05:40:44 localhost kernel: ffff8807b2f97d38 0000000000000086 0000000000000000 ffff88080e035800
Aug  5 05:40:44 localhost kernel: 000000000001ffff ffff88073ad99c80 0000311dfa4681b8 ffff8807fc9df2a0
Aug  5 05:40:44 localhost kernel: 00088002b2f97ca0 000000010333922f ffff8807fdfb25f8 ffff8807b2f97fd8
Aug  5 05:40:44 localhost kernel: Call Trace:
Aug  5 05:40:44 localhost kernel: [<ffffffffa06ace5d>] cv_wait_common+0x11d/0x130 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffff810a1460>] ? autoremove_wake_function+0x0/0x40
Aug  5 05:40:44 localhost kernel: [<ffffffffa06acec5>] __cv_wait+0x15/0x20 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07a9f8c>] zil_commit+0x8c/0x830 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffff815399de>] ? mutex_lock+0x1e/0x50
Aug  5 05:40:44 localhost kernel: [<ffffffffa07491d2>] ? rrw_enter_read_impl+0xd2/0x180 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa079dba2>] zfs_fsync+0x92/0x120 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07b6350>] zpl_fsync+0x60/0xa0 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffff811c45f1>] vfs_fsync_range+0xa1/0x100
Aug  5 05:40:44 localhost kernel: [<ffffffff811c46bd>] vfs_fsync+0x1d/0x20
Aug  5 05:40:44 localhost kernel: [<ffffffff811c46fe>] do_fsync+0x3e/0x60
Aug  5 05:40:44 localhost kernel: [<ffffffff811c4750>] sys_fsync+0x10/0x20
Aug  5 05:40:44 localhost kernel: [<ffffffff8100b0d2>] system_call_fastpath+0x16/0x1b
Aug  5 05:40:44 localhost kernel: INFO: task vim:17171 blocked for more than 600 seconds.
Aug  5 05:40:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:40:44 localhost kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Aug  5 05:40:44 localhost kernel: vim           D 0000000000000005     0 17171  17014 0x00000000
Aug  5 05:40:44 localhost kernel: ffff88055da9bd38 0000000000000082 000000000002d000 ffff88080e035800
Aug  5 05:40:44 localhost kernel: 000000000001ffff ffff880585dcd680 0000000000020000 ffff8807fc9df2a0
Aug  5 05:40:44 localhost kernel: 000280025da9bd18 0000000000001000 ffff88080b13c5f8 ffff88055da9bfd8
Aug  5 05:40:44 localhost kernel: Call Trace:
Aug  5 05:40:44 localhost kernel: [<ffffffff810a16ae>] ? prepare_to_wait_exclusive+0x4e/0x80
Aug  5 05:40:44 localhost kernel: [<ffffffffa06ace5d>] cv_wait_common+0x11d/0x130 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffff810a1460>] ? autoremove_wake_function+0x0/0x40
Aug  5 05:40:44 localhost kernel: [<ffffffffa06acec5>] __cv_wait+0x15/0x20 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07a9f8c>] zil_commit+0x8c/0x830 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffff815399de>] ? mutex_lock+0x1e/0x50
Aug  5 05:40:44 localhost kernel: [<ffffffffa07491d2>] ? rrw_enter_read_impl+0xd2/0x180 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa079dba2>] zfs_fsync+0x92/0x120 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07b6350>] zpl_fsync+0x60/0xa0 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffff811c45f1>] vfs_fsync_range+0xa1/0x100
Aug  5 05:40:44 localhost kernel: [<ffffffff811c46bd>] vfs_fsync+0x1d/0x20
Aug  5 05:40:44 localhost kernel: [<ffffffff811c46fe>] do_fsync+0x3e/0x60
Aug  5 05:40:44 localhost kernel: [<ffffffff811c4750>] sys_fsync+0x10/0x20
Aug  5 05:40:44 localhost kernel: [<ffffffff8100b0d2>] system_call_fastpath+0x16/0x1b
Aug  5 05:40:44 localhost kernel: INFO: task zfs:29181 blocked for more than 600 seconds.
Aug  5 05:40:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:40:44 localhost kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Aug  5 05:40:44 localhost kernel: zfs           D 0000000000000005     0 29181  29158 0x00000000
Aug  5 05:40:44 localhost kernel: ffff88053ac33a18 0000000000000086 0000000000000000 ffffffffa07064f0
Aug  5 05:40:44 localhost kernel: ffff8807fe0f4ff8 0000000000000000 000031172e32a9a6 ffffffffa0723f96
Aug  5 05:40:44 localhost kernel: ffff8808084f6768 0000000103332085 ffff88057750e5f8 ffff88053ac33fd8
Aug  5 05:40:44 localhost kernel: Call Trace:
Aug  5 05:40:44 localhost kernel: [<ffffffffa07064f0>] ? dbuf_rele+0x40/0x50 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa0723f96>] ? dnode_hold_impl+0x226/0x500 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa06ace5d>] cv_wait_common+0x11d/0x130 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffff810a1460>] ? autoremove_wake_function+0x0/0x40
Aug  5 05:40:44 localhost kernel: [<ffffffff815399de>] ? mutex_lock+0x1e/0x50
Aug  5 05:40:44 localhost kernel: [<ffffffffa06acec5>] __cv_wait+0x15/0x20 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07641df>] txg_wait_synced+0xef/0x140 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa06add39>] ? tsd_hash_dtor+0x79/0x90 [spl]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07aa5cb>] zil_commit+0x6cb/0x830 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07ac20c>] zil_suspend+0x22c/0x2b0 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa07280c1>] dsl_dataset_snapshot+0x1d1/0x2b0 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa02d5190>] ? avl_find+0x60/0xb0 [zavl]
Aug  5 05:40:44 localhost kernel: [<ffffffffa0760ef3>] ? spa_lookup+0x53/0x60 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa075877b>] ? spa_open_common+0x29b/0x440 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa0748e45>] ? rrw_exit+0x65/0x180 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa06cb454>] ? nvlist_lookup_common+0x84/0xd0 [znvpair]
Aug  5 05:40:44 localhost kernel: [<ffffffffa078ec61>] zfs_ioc_snapshot+0x261/0x290 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffffa078eeff>] zfsdev_ioctl+0x1cf/0x4d0 [zfs]
Aug  5 05:40:44 localhost kernel: [<ffffffff811a7312>] vfs_ioctl+0x22/0xa0
Aug  5 05:40:44 localhost kernel: [<ffffffff811a74b4>] do_vfs_ioctl+0x84/0x580
Aug  5 05:40:44 localhost kernel: [<ffffffff811a7a31>] sys_ioctl+0x81/0xa0
Aug  5 05:40:44 localhost kernel: [<ffffffff8100b0d2>] system_call_fastpath+0x16/0x1b
====================================================]

3) Meanwhile, the physical devices keep responding normally, as evidenced by smartd logging (as I mentioned above, they also responded normally to dd'ing large areas of them to /dev/null): 
====================================================[
Aug  5 05:41:08 localhost smartd[4754]: Device: /dev/sda [SAT], SMART Usage Attribute: 194 Temperature_Celsius changed from 139 to 200
Aug  5 05:41:09 localhost smartd[4754]: Device: /dev/sda [SAT], old test of type S not run at Fri Aug  5 02:00:00 2016 BRT, starting now.
Aug  5 05:41:10 localhost smartd[4754]: Device: /dev/sda [SAT], starting scheduled Short Self-Test.
Aug  5 05:41:11 localhost smartd[4754]: Device: /dev/sdb [SAT], SMART Prefailure Attribute: 3 Spin_Up_Time changed from 190 to 191
Aug  5 05:41:11 localhost smartd[4754]: Device: /dev/sdb [SAT], SMART Usage Attribute: 194 Temperature_Celsius changed from 106 to 120
Aug  5 05:41:11 localhost smartd[4754]: Device: /dev/sdb [SAT], old test of type S not run at Fri Aug  5 02:00:00 2016 BRT, starting now.
Aug  5 05:41:11 localhost smartd[4754]: Device: /dev/sdb [SAT], starting scheduled Short Self-Test.
Aug  5 05:41:11 localhost smartd[4754]: Device: /dev/sdc [SAT], old test of type S not run at Fri Aug  5 02:00:00 2016 BRT, starting now.
Aug  5 05:41:11 localhost smartd[4754]: Device: /dev/sdc [SAT], starting scheduled Short Self-Test.
====================================================]

4) After some more "task ... blocked" logs for txg_sync, AioMgr13-F and vim very similar to the above, I got this one (I investigated and discovered that EMT is, apparently, a LWP task [thread?] running inside a VirtualBox process):
====================================================[
Aug  5 05:50:44 localhost kernel: INFO: task EMT:17660 blocked for more than 600 seconds.
Aug  5 05:50:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:50:44 localhost kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Aug  5 05:50:44 localhost kernel: EMT           D 0000000000000006     0 17660   6107 0x00000000
Aug  5 05:50:44 localhost kernel: ffff8807fafb3d38 0000000000000086 00000000000001d8 ffff88080e035800
Aug  5 05:50:44 localhost kernel: 000000000001ffff ffff88073ad99b80 0000000000020000 ffff8807fc9df2a0
Aug  5 05:50:44 localhost kernel: 00088002fafb3d88 0000000000000190 ffff8807fdef3ad8 ffff8807fafb3fd8
Aug  5 05:50:44 localhost kernel: Call Trace:
Aug  5 05:50:44 localhost kernel: [<ffffffff810a16ae>] ? prepare_to_wait_exclusive+0x4e/0x80
Aug  5 05:50:44 localhost kernel: [<ffffffffa06ace5d>] cv_wait_common+0x11d/0x130 [spl]
Aug  5 05:50:44 localhost kernel: [<ffffffff810a1460>] ? autoremove_wake_function+0x0/0x40
Aug  5 05:50:44 localhost kernel: [<ffffffffa06acec5>] __cv_wait+0x15/0x20 [spl]
Aug  5 05:50:44 localhost kernel: [<ffffffffa07a9f8c>] zil_commit+0x8c/0x830 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffff815399de>] ? mutex_lock+0x1e/0x50
Aug  5 05:50:44 localhost kernel: [<ffffffffa07491d2>] ? rrw_enter_read_impl+0xd2/0x180 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa079dba2>] zfs_fsync+0x92/0x120 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa07b6350>] zpl_fsync+0x60/0xa0 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffff811c45f1>] vfs_fsync_range+0xa1/0x100
Aug  5 05:50:44 localhost kernel: [<ffffffff811c46bd>] vfs_fsync+0x1d/0x20
Aug  5 05:50:44 localhost kernel: [<ffffffff811c46fe>] do_fsync+0x3e/0x60
Aug  5 05:50:44 localhost kernel: [<ffffffff811c4750>] sys_fsync+0x10/0x20
Aug  5 05:50:44 localhost kernel: [<ffffffff8100b0d2>] system_call_fastpath+0x16/0x1b
====================================================]

5) Then, naturally, other ZFS tasks start locking up too:
====================================================[
Aug  5 05:50:44 localhost kernel: INFO: task zfs:29181 blocked for more than 600 seconds.
Aug  5 05:50:44 localhost kernel:      Tainted: P           -- ------------    2.6.32-573.7.1.el6.x86_64 #1
Aug  5 05:50:44 localhost kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
Aug  5 05:50:44 localhost kernel: zfs           D 0000000000000005     0 29181  29158 0x00000000
Aug  5 05:50:44 localhost kernel: ffff88053ac33a18 0000000000000086 0000000000000000 ffffffffa07064f0
Aug  5 05:50:44 localhost kernel: ffff8807fe0f4ff8 0000000000000000 000031172e32a9a6 ffffffffa0723f96
Aug  5 05:50:44 localhost kernel: ffff8808084f6768 0000000103332085 ffff88057750e5f8 ffff88053ac33fd8
Aug  5 05:50:44 localhost kernel: Call Trace:
Aug  5 05:50:44 localhost kernel: [<ffffffffa07064f0>] ? dbuf_rele+0x40/0x50 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa0723f96>] ? dnode_hold_impl+0x226/0x500 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa06ace5d>] cv_wait_common+0x11d/0x130 [spl]
Aug  5 05:50:44 localhost kernel: [<ffffffff810a1460>] ? autoremove_wake_function+0x0/0x40
Aug  5 05:50:44 localhost kernel: [<ffffffff815399de>] ? mutex_lock+0x1e/0x50
Aug  5 05:50:44 localhost kernel: [<ffffffffa06acec5>] __cv_wait+0x15/0x20 [spl]
Aug  5 05:50:44 localhost kernel: [<ffffffffa07641df>] txg_wait_synced+0xef/0x140 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa06add39>] ? tsd_hash_dtor+0x79/0x90 [spl]
Aug  5 05:50:44 localhost kernel: [<ffffffffa07aa5cb>] zil_commit+0x6cb/0x830 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa07ac20c>] zil_suspend+0x22c/0x2b0 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa07280c1>] dsl_dataset_snapshot+0x1d1/0x2b0 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa02d5190>] ? avl_find+0x60/0xb0 [zavl]
Aug  5 05:50:44 localhost kernel: [<ffffffffa0760ef3>] ? spa_lookup+0x53/0x60 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa075877b>] ? spa_open_common+0x29b/0x440 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa0748e45>] ? rrw_exit+0x65/0x180 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa06cb454>] ? nvlist_lookup_common+0x84/0xd0 [znvpair]
Aug  5 05:50:44 localhost kernel: [<ffffffffa078ec61>] zfs_ioc_snapshot+0x261/0x290 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffffa078eeff>] zfsdev_ioctl+0x1cf/0x4d0 [zfs]
Aug  5 05:50:44 localhost kernel: [<ffffffff811a7312>] vfs_ioctl+0x22/0xa0
Aug  5 05:50:44 localhost kernel: [<ffffffff811a74b4>] do_vfs_ioctl+0x84/0x580
Aug  5 05:50:44 localhost kernel: [<ffffffff811a7a31>] sys_ioctl+0x81/0xa0
Aug  5 05:50:44 localhost kernel: [<ffffffff8100b0d2>] system_call_fastpath+0x16/0x1b
====================================================]

6) Here's my current "zpool status" output (taken with the system in the above state):
====================================================[
zpool status
  pool: pool01
 state: ONLINE
status: The pool is formatted using a legacy on-disk format.  The pool can
        still be used, but some features are unavailable.
action: Upgrade the pool using 'zpool upgrade'.  Once this is done, the
        pool will no longer be accessible on software that does not support
        feature flags.
  scan: scrub repaired 0 in 4h2m with 0 errors on Tue Aug  2 07:48:23 2016
config:
        NAME                                                   STATE     READ WRITE CKSUM
        pool01                                                 ONLINE       0     0     0
          mirror-0                                             ONLINE       0     0     0
            sdb                                                ONLINE       0     0     0
            sda1                                               ONLINE       0     0     0
        logs
          sdc6                                                 ONLINE       0     0     0
        cache
          scsi-SATA_M4-CT256M4SSD3_00000000124503532F7A-part5  ONLINE       0     0     0
errors: No known data errors
====================================================]

Finally, I want to note again that the pool seems to be OK for reading: I just finished a "find -ls" on it, and it finished with no apparent issues after traversing the entire pool's main dataset.

I will keep the machine up for a few more hours, to update my backups taking advantage that the pool doesn't seem to have any issues with reads. If I can assist with anything else, now is the time to ask for it... 

Cheers, 
 -- 
   Durval.

**Comment by GeLiXin** [comment]
It seems your pool is suspended, have you tried : zpool clear pool01 ?

**Comment by DurvalMenezes** [comment]
Hello @GeLiXin, 

> It seems your pool is suspended, 

Humrmrm... if it was, wouldn't it say so on "zpool status"? 

> have you tried : zpool clear pool01 ?

I've just finished updating my backups here (around 120GB of it, much more than I estimated at first, but it all completed OK and with no issues with reading it all from the pool), then I tried the "zpool clear pool01" you suggested; After a few minutes with the command just hanging there (neither finishing, nor spitting any messages), I checked "ps auwx"  and sure enough, it's there hanging in 'D' state like every other process that has had anything with writing to the pool:

   root     12383  0.0  0.0 128216  1488 pts/24   D+   09:57   0:00 zpool clear pool01

So I'm rebooting the machine now. I hope it comes back with the pool in an usable state and I don't have to restore everything from those backups... 

Cheers, 
-- 
   Durval.

**Comment by DurvalMenezes** [comment]
Hello, 

During reboot, system locked up when trying to umount the ZFS datasets (as expected, I guess), so I had to power-cycle it. Thankfully, after a really long (many minutes) and unnerving "zfs mount -a" during system initialization. everything seems to have come up Ok. Seems I won't need those backups after all, at least not now ("the only backups you need are the ones you didn't make").

If there's anything I can do to help diagnose and fix this issue, please let me know. 

Thanks, 
-- 
   Durval.
